contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -714,6 +714,128 @@ $BODY$
 STRICT
 LANGUAGE SQL IMMUTABLE;
 
+CREATE OR REPLACE FUNCTION sys.DATETIMEOFFSETFROMPARTS(IN p_year NUMERIC,
+                                                               IN p_month NUMERIC,
+                                                               IN p_day NUMERIC,
+                                                               IN p_hour NUMERIC,
+                                                               IN p_minute NUMERIC,
+                                                               IN p_seconds NUMERIC,
+                                                               IN p_fractions NUMERIC,
+                                                               IN p_hour_offset NUMERIC,
+                                                               IN p_minute_offset NUMERIC,
+                                                               IN p_precision NUMERIC)
+RETURNS sys.DATETIMEOFFSET
+AS
+$BODY$
+DECLARE
+    v_err_message VARCHAR;
+    v_fractions VARCHAR;
+    v_precision SMALLINT;
+    v_calc_seconds NUMERIC; 
+    v_resdatetime TIMESTAMP WITHOUT TIME ZONE;
+    v_string TEXT;
+    v_sign TEXT;
+BEGIN
+    v_fractions := floor(p_fractions)::INTEGER::VARCHAR;
+    v_precision := p_precision::SMALLINT;
+
+    IF (scale(p_precision) > 0) THEN
```

Comment:
What happens if `p_precision` is null? According to the official document, it should raise an error.